.github/workflows/pr-checks.yml

```
@@ -20,9 +20,12 @@ jobs:
 
       - name: "Check for changes in uv.lock"
         run: |
-          # fail if uv sync had to change uv.lock
-          uv sync
-          git diff --exit-code uv.lock
+           uv sync
+            if ! git diff --quiet uv.lock; then
+              echo "uv.lock has changed. Run 'uv sync' locally and commit the updated lock file."
+              git diff uv.lock
+              exit 1  # Fail the job
+            fi
```

```suggestion
          uv sync
          if ! git diff --quiet uv.lock; then
            echo "uv.lock has changed. Run 'uv sync' locally and commit the updated lock file."
            git diff uv.lock
            exit 1  # Fail the job
          fi
```

(just a surprising indent level, works either way :))